crates/engine/src/ext.rs

```diff
@@ -417,6 +419,39 @@ impl Engine {
             "off-chain environment does not yet support `call_chain_extension`"
         );
     }
+
+    pub fn ecdsa_recover(
+        &mut self,
+        signature: &[u8; 65],
+        message_hash: &[u8; 32],
+        output: &mut [u8; 33],
+    ) -> Result {
+        use secp256k1::{
+            recover,
+            Message,
+            RecoveryId,
+            Signature,
+        };
+
+        let recovery_byte = if signature[64] > 27 {
```

Comment:
Something that I'm missing is an explanation of this magic number. How about creating a `const` with a mnemonic name and adding a comment to that `const`?

Reply:
Please turn the `27` into a `const` with proper documentation of where it comes. Ideally a link.

Reply:
I added the comment before the `if` section. It is more clear than a comment near a constant. This value is used in several crates, so I can't create one global constant. Define it in every crate seems overkill and better to leave the comment because `27` is used in one place.